agent/engine/docker_container_engine.go

```diff
@@ -160,6 +162,55 @@ func (dg *dockerGoClient) dockerClient() (dockeriface.Client, error) {
 	return dg.clientFactory.GetClient(dg.version)
 }
 
+func (dg *dockerGoClient) LoadImage(name string, reader io.ReadCloser) DockerContainerMetadata {
+	timeout := ttime.After(loadImageTimeout)
+	pullLock.Lock()
+	defer pullLock.Unlock()
+
+	response := make(chan DockerContainerMetadata, 1)
+
+	go func() { response <- dg.loadImage(name, reader) }()
+
+	select {
+	case resp := <-response:
+		return resp
+	case <-timeout:
+		return DockerContainerMetadata{Error: &DockerTimeoutError{loadImageTimeout, "loaded"}}
+	}
+}
+
+func (dg *dockerGoClient) loadImage(name string, reader io.ReadCloser) DockerContainerMetadata {
+	log.Debug("Loading image", "name", name)
+	defer reader.Close()
+	client, err := dg.dockerClient()
+	if err != nil {
+		return DockerContainerMetadata{Error: CannotGetDockerClientError{version: dg.version, err: err}}
+	}
+
+	_, err = client.InspectImage(name)
```

Comment:
This seems to imply that once an image is loaded, it can never be refreshed.  I have two objections here:

1. The existing pull logic is such that any time a new task is started, all the container images in the task are pulled.  This lets an update pushed to the registry actually run on the container instance.
2. Logic for deciding whether or not to load shouldn't live in this layer of the code, but rather one layer up where it gets called.  The `dockerGoClient` struct is really just wrapping the client and providing timeouts + error messaging.

I think what you're ultimately trying to avoid here is downloading from S3 every time.  I think that's a good goal, but can be accomplished by saving the [`ETag`](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html) from S3 and checking whether the `ETag` has changed.